home/tests.py

```diff
@@ -205,3 +208,149 @@ def tags_response(self, client):
+            # User did'nt choose grade
+            ((10, 1, "Question in Math", 'How much is it 1+1?',
+             datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC), 1, 4, Grade.GRADE7, 1, -10, False), ValidationError),
+            # User chose negative book number
+            ((10, 1, "Question in Math", 'How much is it 1+1?',
+             datetime(2022, 4, 7, 12, 53, 29, 4, tzinfo=pytz.UTC),
+             1, 4, Grade.GRADE7, 1, 99990, False), ValidationError),
+            # User entered to high book number
+        ])
+        # checks if the question was not inserted
+        @pytest.mark.django_db
+        def test_add_invalid_question(self, invalid_Input, exception):
+
+            with pytest.raises(exception):
+                assert Question(*invalid_Input).clean_fields()
+
+        # checks default value in is_edited
+        @pytest.mark.django_db
+        def test_default_is_edited(self, question):
+
+            assert question.is_edited is False
+
+        # checks default value in book_page once the user didn't insert a book page
+        @pytest.mark.django_db
+        def test_default_book_page(self, question):
+
+            assert question.book_page is None
+
+        # checks default value in book once the user didn't insert a book
+        @pytest.mark.django_db
+        def test_default_book(self, question):
+
+            assert question.book is None
+
+        # checks default value in sub_subject once the user didn't insert a sub-subject
+        @pytest.mark.django_db
+        def test_default_sub_subject(self, question):
+
+            assert question.sub_subject is None
+
+    class TestViews:
+        @pytest.mark.parametrize("valid_data", [
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7}),
+            # User entered all the required fields
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7, 'sub-subject': 2}),
+            # User entered all the required fields and sub-subject
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7, 'book': 2}),
+            # User entered all the required fields and sub-subject and book
+            ({'title': "Question in Math", 'content': 'How much is it 1+1?',
+              'subject': 1, 'grade': Grade.GRADE7, 'book_page': 23}),
+            # User entered all the required fields and a valid book page
+        ])
+        @pytest.mark.django_db
+        def test_post_valid_question_with_client(self, client, valid_data):
+            latestInsertedQuestionID = Question.objects.latest('id').id
+            client.login(username='Lior', password='LiorLior')
+            client.post('/explore/new_question', data=valid_data)
+
+            assert Question.objects.latest('id').id != latestInsertedQuestionID
```

Comment:
this check is insufficient, how do you know that data you;ve posted actually made it to the DB?